frontend/src/api/article.ts

```diff
@@ -0,0 +1,12 @@
+import { AllArticleResponseType } from '@/types/article';
+import { instance } from '.';
+
+export const getTotalArticles = async ({ pageNumber }: { pageNumber: number }) => {
+  const { data } = await instance.get<AllArticleResponseType>(`/article?pagenumber=${pageNumber}`);
+  return data;
+};
+
+export const getTotalArticleCount = async () => {
+  const { data } = await instance.get<{ count: number }>('/article/total-count');
+  return data;
```

Comment:
count만 넘기는거라면 함수 자체에서도 count를 가져온다라고 되어있으니 `{ count: number }`로 넘기는 것보다 `number`로 가공해서 넘기는 건 어떻게 생각하시나유?

한다면 then 문 안에서 data.count만 return하는 형식이 될 것 같네용

Reply:
리액트 쿼리에서 https://tanstack.com/query/v4/docs/react/reference/useQuery 
queryFn에 넘어오는 함수는 Promise를 return 해야한다고 읽어서 data를 넘기도록 처리하였습니다! data.count만 return 해도 작동이 되는걸까요? 시도를 안해봐서 궁금하네요 🤔

Reply:
저는 샐리처럼 await을 사용하지 않고 Promise 문 자체를 리턴하는 방식을 사용하였는데 잘 작동했습니다.

```suggestion
  return instance.get<{ count: number }>('/article/total-count').then(res => res.data.count);
```

Reply:
저도 Promise로 직접 리턴하는게 더 깔끔해보이네요!

Reply:
생각해보니 await로 처리하지 않아도 then처리를 하면 Promise가 반환되니깐 가능하겠네요

Reply:
그런데 해당 부분의 api는 제거예정이라 getTotalArticle에 적용해보겠습니다